internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -49,6 +49,14 @@ func createMountsConfig(ctx context.Context, coi *createOptionsInternal) (*mount
 			src, dst := uvm.GetContainerPipeMapping(coi.HostingSystem, mount)
 			config.mpsv1 = append(config.mpsv1, schema1.MappedPipe{HostPath: src, ContainerPipeName: dst})
 			config.mpsv2 = append(config.mpsv2, hcsschema.MappedPipe{HostPath: src, ContainerPipeName: dst})
+		} else if strings.HasPrefix(mount.Source, guestpath.UVMMountPrefix) {
+			if uvm.IsPipe(strings.TrimPrefix(mount.Source, guestpath.UVMMountPrefix)) {
+				src, dst := uvm.GetContainerPipeMapping(coi.HostingSystem, mount)
+				config.mpsv1 = append(config.mpsv1, schema1.MappedPipe{HostPath: src, ContainerPipeName: dst})
+				config.mpsv2 = append(config.mpsv2, hcsschema.MappedPipe{HostPath: src, ContainerPipeName: dst})
+			} else {
+				return nil, fmt.Errorf("unsupported UVM mount source: %s", mount.Source)
+			}
```

Comment:
Can this logic be consolidated somehow? L54-56 are the same, and its just the checks that are different, right?

Also, here, `GetContainerPipeMapping`, and `UVMNamedPipe` all repeatedly do different variations of `Trim`/`CutPrefix(guestpath.UVMMountPrefix|pipePrefix)`

maybe we need a `PipeType` type to parse the string once and return what it is?